### PR TITLE
feat(components/atom/tooltip): functional component

### PR DESCRIPTION
**Pull request by andresin87**
## ATOM/TOOLTIP
**TASK**: 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation
- [x] Refactor

### Description, Motivation and Context

- Functional Components
- title & content --> content
- HoC removed:
  - withIntersectionObserver
  - withOpenToogle
- Hooks usage
  - [useControlledState](https://github.com/SUI-Components/adevinta-spain-components/tree/master/components/react/hooks#usecontrolledstate) (adevinta-spain-components) (@s-ui/react-hooks)
  - [useIntersection](https://github.com/streamich/react-use/blob/master/docs/useIntersection.md) (react-use)
- removes the resulting color classes cached.

### Inspiration
- ref target
  - [reactstrap 8.9.0](https://reactstrap.github.io/components/tooltips/#app)
- wrapping
  - [carbon](https://www.carbondesignsystem.com/components/tooltip/usage)
  - [chakra](https://chakra-ui.com/docs/overlay/tooltip)
  - [material-ui](https://material-ui.com/components/tooltips/#tooltip)
  - [tooltip](https://ant.design/components/tooltip/#header)

### CODE REVIEW

File: components/atom/tooltip/src/index.scss [hunk 1]
```diff
@@ -221,6 +200,20 @@ $c-atom-tooltip-colors: (
     }
   }
 
+  &[data-placement='left-start'],
+  &[data-placement='right-start'] {
+    #{$class-inner} {
+      transform: translateY(-8px);
```
Comment (nistadev) [inline components/atom/tooltip/src/index.scss]: It is better to use variables for this `translateY` value.

File: components/atom/tooltip/src/index.scss [hunk 2]
```diff
@@ -221,6 +200,20 @@ $c-atom-tooltip-colors: (
...
+    }
+  }
+
+  &[data-placement='left-end'],
+  &[data-placement='right-end'] {
+    #{$class-inner} {
+      transform: translateY(8px);
```
Comment (nucliweb) [inline components/atom/tooltip/src/index.scss]: Same here

File: components/atom/tooltip/src/index.scss (hunk 1 above)
Comment (andresin87) [reply components/atom/tooltip/src/index.scss]: removed – deprecated selectors

File: components/atom/tooltip/src/index.scss (hunk 2 above)
Comment (andresin87) [reply components/atom/tooltip/src/index.scss]: same (removed – deprecated selectors)

File: components/atom/tooltip/src/AtomTooltip.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+const AtomTooltip = forwardRef(
+  (
+    {
+      children,
+      color,
+      content,
+      defaultIsVisible,
+      delay,
+      isArrowed = true,
+      isVisible,
```
Comment (nucliweb) [inline components/atom/tooltip/src/AtomTooltip.js:34]: As you were commented in the SUI Components Weekly today, we would define the default value to `true`, right?

```suggestion
      isVisible = true,
```

Comment (andresin87) [reply components/atom/tooltip/src/AtomTooltip.js:34]: default naming:
- if bolean prop --> prefix is (no with)
- semantic affirmative (visible – no hidden)

resulting attribute name: `isVisible`

the default value depends in the behaviour desired for the component: in this case `undefined`

another example:
- hidearrow --> isArrowed
- withLabel --> isLabeled (this example opens the next point -with is a reserved prefix used for HoC and use is reserved for Hooks)

**Review by nucliweb — APPROVED** [review]

File: components/atom/tooltip/src/index.scss [hunk 4]
```diff
@@ -82,7 +88,7 @@ $c-atom-tooltip-colors: (
 }
 
 #{$base-class} {
-  display: block;
+  display: contents;
```
Comment (nucliweb) [inline components/atom/tooltip/src/index.scss:91]: 👀  The `display: contents` don't have support for IE 11, and has partial support for Safari, Safari iOS, and Samsung Internet, take a look at the [support](https://caniuse.com/css-display-contents)

Comment (andresin87) [reply components/atom/tooltip/src/index.scss:91]: https://github.com/SUI-Components/sui-components/blob/master/components/atom/popover/src/index.scss#L33

2 months ago (nobody has reported any issue)

Comment (andresin87) [reply components/atom/tooltip/src/index.scss:91]: 🤷

**Review by nistadev — APPROVED** [review]

**Pull request merged**